app/api/vercel/webhook/route.ts

```diff
@@ -0,0 +1,219 @@
+    
+    if (!vercelSignature) {
+      console.error("Missing x-vercel-signature header");
+      return NextResponse.json(
+        { error: "Missing x-vercel-signature header" },
+        { status: 400 }
+      );
+    }
+
+    const hmac = createHmac("sha1", vercelWebhookSecret);
+    hmac.update(body);
+    const expectedSignature = hmac.digest("hex");
+    
+    if (vercelSignature !== expectedSignature) {
+      console.error("Vercel webhook signature verification failed");
+      return NextResponse.json(
+        { error: "Webhook signature verification failed" },
+        { status: 400 }
+      );
+    }
+
+    const payload = JSON.parse(body) as VercelWebhookPayload;
+    const eventType = payload.type;
+
+    console.log(`Received Vercel webhook event: ${eventType}`);
+
+    if (eventType === "deployment.error") {
+      await handleDeploymentError(payload);
+    } else {
+      console.log(`Unhandled event type: ${eventType}`);
+    }
+
+    return NextResponse.json({ received: true });
+  } catch (error) {
+    console.error("Error processing Vercel webhook:", error);
+    return NextResponse.json(
+      { error: "Failed to process Vercel webhook" },
+      { status: 500 }
+    );
+  }
+}
+
+async function handleDeploymentError(payload: VercelWebhookPayload) {
+  console.log("Deployment error:", {
+    deploymentId: payload.payload.deployment.id,
+    deploymentUrl: payload.payload.deployment.url,
+    dashboardLink: payload.payload.links.deployment,
+    target: payload.payload.target
+  });
+
+  try {
+    const prNumber = extractPRNumber(payload.payload.deployment.meta);
+    
+    if (!prNumber) {
+      console.log("No PR number found in deployment metadata, skipping PR comment");
+      return;
+    }
+
+    const buildLogs = await fetchVercelBuildLogs(payload.payload.deployment.id);
+    
+    const comment = formatBuildLogsComment(payload, buildLogs);
+    
+    await postGitHubComment(
+      comment,
+      "vargasjr-dev-vercel-webhook/1.0",
+      "Posted Vercel build logs to PR"
+    );
+
+    console.log(`Successfully posted build logs to PR #${prNumber}`);
+    
+  } catch (error) {
+    console.error("Error handling deployment error:", error);
+    throw error;
+  }
+}
+
+function extractPRNumber(meta: Record<string, string>): string | null {
+  const prNumber = meta.githubPullRequestId || 
+                   meta.gitlabMergeRequestId || 
+                   meta.pullRequestId ||
+                   meta.prNumber;
+  
+  if (prNumber) {
+    return prNumber;
+  }
+
+  const branchName = meta.githubCommitRef || meta.gitBranch || meta.branch;
+  if (branchName) {
+    const prMatch = branchName.match(/pr[/-](\d+)/i);
+    if (prMatch) {
+      return prMatch[1];
+    }
+  }
+
+  return null;
+}
+
+async function fetchVercelBuildLogs(deploymentId: string): Promise<string> {
+  const vercelToken = process.env.VERCEL_TOKEN;
+  const vercelTeamId = process.env.VERCEL_TEAM_ID;
```

Comment:
We can hardcode the team if as a top of the file constant: team_36iZPJkU2LLMsHZqJZXMZppe